### PR TITLE
Use cluster/log-dump.sh to collect base cluster logs in kubemark

### LINKED ISSUE #24415: When Kubemark tests fail JUnit_XX files are not filled correctly.
Even when some test fails contents are:
```
<testsuite tests="0" failures="0" time="15.245789696">
...
```

cc @kubernetes/sig-testing @wojtek-t

### PR DESCRIPTION
**Pull request by ixdy**
Fixes #24415.

cc @fejta @spxtr @wojtek-t

### CODE REVIEW

File: hack/jenkins/e2e-runner.sh [hunk 1]
```diff
@@ -137,6 +137,14 @@ function install_google_cloud_sdk_tarball() {
     export PATH=${install_dir}/google-cloud-sdk/bin:${PATH}
 }
 
+function dump_cluster_logs_and_exit() {
+    local -r exit_status=$1
```
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: Nit: might be worth just making this equal to `$?` and getting rid of the argument.

**Comment by spxtr** [comment]
Nit or not, LGTM.

File: hack/jenkins/e2e-runner.sh (hunk 1 above)
Comment (ixdy) [inline hack/jenkins/e2e-runner.sh]: good suggestion. I wrote this bit before I replaced all the if statements, but I like it.

**Comment by wojtek-t** [comment]
Thanks @ixdy - LGTM

**Comment by gmarek** [comment]
Travis problems... #24628

**Pull request merged**